development/compiling/optimizing_for_size.rst

```diff
@@ -3,49 +3,195 @@
 Optimizing a build for size
 ===========================
 
-.. highlight:: shell
+.. highlight:: none
 
 Rationale
 ---------
 
 Sometimes, it is desired to optimize a build for size rather than speed.
 This means not compiling unused functions from the engine, as well as using
 specific compiler flags to aid on decreasing build size.
-Common situations include creating builds for mobile and Web platforms.
+Common situations include creating builds for mobile and web platforms.
 
 This tutorial aims to give an overview on different methods to create
 a smaller binary. Before continuing, it is recommended to read the previous tutorials
 on compiling Godot for each platform.
 
-.. seealso::
-
-    You can use the online
-    `Godot build options generator <https://godot-build-options-generator.github.io/>`__
-    to generate a ``custom.py`` file containing SCons options.
-    You can then save this file and place it at the root of your Godot source directory.
-
 Disabling 3D
 ------------
 
-For 2D games, having the whole 3D engine available usually makes no sense. Because of this, there is a build flag to disable it:
+For 2D games, having the whole 3D engine available usually makes no sense. Because of this,
+there is a build flag to disable most of it. Without the 3D parts, the binary size can be
+reduced by about 15%:
 
 ::
 
-    scons p=windows target=release tools=no disable_3d=yes
-
-Tools must be disabled in order to use this flag, as the editor is not designed
-to operate without 3D support. Without it, the binary size can be reduced
-by about 15%.
+    scons p=<platform> target=release tools=no disable_3d=yes
+
+The following classes are not available when building with ``disable_3d=yes``:
+
+- :ref:`class_ARVRAnchor`
+- :ref:`class_ARVRCamera`
+- :ref:`class_ARVRController`
+- :ref:`class_ARVROrigin`
+- :ref:`class_AnimatedSprite3D`
+- :ref:`class_Area`
+- :ref:`class_AudioStreamPlayer3D`
```

Comment:
I think maintaining this list might end up being a hassle. The list of modules that can be disabled was already regularly outdated, so the same fate will likely meet this one.

I think it would be better if this was generated on CI for https://godot-build-options-generator.github.io/ so that the information can be displayed there when users want to disable 3D or advanced GUI nodes.